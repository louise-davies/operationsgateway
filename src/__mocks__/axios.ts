// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

const requests = {
  get: jest.fn((path) => {
    if (path === '/operationsgateway-settings.json') {
      return Promise.resolve({
        data: {
          apiUrl: 'api',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
};

export default requests;
