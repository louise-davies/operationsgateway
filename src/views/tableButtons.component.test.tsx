import React from 'react';
import TableButtons from './tableButtons.component';
import { render, screen } from '@testing-library/react';
import axios from 'axios';
import userEvent from '@testing-library/user-event';

describe('Table buttons', () => {
  const openFilters = jest.fn();
  const openChannels = jest.fn();
  const toggleSearchExpanded = jest.fn();

  let user;
  let props: React.ComponentProps<typeof TableButtons>;

  const createView = () => {
    return render(<TableButtons {...props} />);
  };

  beforeEach(() => {
    user = userEvent.setup();
    (axios.get as jest.Mock).mockResolvedValue({ data: [] });
    props = {
      openFilters,
      openChannels,
      toggleSearchExpanded,
      searchExpanded: false,
    };
  });

  it('renders the buttons', async () => {
    const view = createView();
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('calls openFilters when the filters button is clicked', async () => {
    createView();

    await user.click(screen.getByRole('button', { name: 'Filters' }));
    expect(openFilters).toHaveBeenCalled();
  });

  it('calls openChannels when the data channels button is clicked', async () => {
    createView();

    await user.click(screen.getByRole('button', { name: 'Data Channels' }));
    expect(openChannels).toHaveBeenCalled();
  });

  it('calls toggleSearchExpanded when the show/hide search button is clicked', async () => {
    // set this to true to test that the button name changes from the one we have in the snapshot
    props.searchExpanded = true;
    createView();

    await user.click(screen.getByRole('button', { name: 'Hide search' }));
    expect(toggleSearchExpanded).toHaveBeenCalled();
  });
});
