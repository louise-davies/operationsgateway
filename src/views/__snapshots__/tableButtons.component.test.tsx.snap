// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table buttons renders the buttons 1`] = `
<DocumentFragment>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-iajp3t-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Show search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Data Channels
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Filters
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Functions
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
